### PR TITLE
Implement labels show/hide toggle for vector tile layers

### PR DESCRIPTION
**Pull request by nirvn**
In action:

[Screencast from 2023-06-29 12-48-26.webm](https://github.com/opengisch/QField/assets/1728657/899d18d3-0063-4efa-a2a7-c06e7de827a6)

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for dc79ed489985fb1d58345a49299bfcec803e190b: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4407-72008e43eb606a1a48725d4c0e917294349e86fd-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4407-72008e43eb606a1a48725d4c0e917294349e86fd-arm-neon-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4407-72008e43eb606a1a48725d4c0e917294349e86fd-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4407-72008e43eb606a1a48725d4c0e917294349e86fd-x86-android.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4407-72008e43eb606a1a48725d4c0e917294349e86fd-arm64-android.apk)
- [all access arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4407-72008e43eb606a1a48725d4c0e917294349e86fd-arm-neon-android.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4407-72008e43eb606a1a48725d4c0e917294349e86fd-x64-android.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4407-72008e43eb606a1a48725d4c0e917294349e86fd-x86-android.apk)
</details>

File: src/core/qfield.h.in [hunk 1]
```diff
@@ -58,6 +58,7 @@ namespace qfield
   static inline const QString appName{ QStringLiteral( "@APP_NAME@" ) };
 }; // namespace qfield
 
+#define WITH_VCPKG @WITH_VCPKG@
```
Comment (m-kuhn) [inline src/core/qfield.h.in:61]: That looks like a workaround. What are you trying to solve?

Comment (nirvn) [reply src/core/qfield.h.in:61]: It's not really a workaround but more of an attempt at leveraging the flexibility of vcpkg's bsckporting of features through patches while not forcing a non-vcpkg minimum QGIS version to be QGIS dev branch. 

With vcpkg, we're in full control of what QGIS is built with. We want to stick to stable 3.32 (which means _QGIS_VERSION would be 332000. But we're pushing a backport through patch, hence the WITH_VCPKG.

That way, we get the best of all worlds.

Comment (m-kuhn) [reply src/core/qfield.h.in:61]: Can we just patch the qgis version number to 332001 instead?

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**